src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Double.valueOf(s).longValue() : Long.valueOf(s);
```

Comment:
@xvrl right , following change should be good enough.
-        return s.contains(".") ? Double.valueOf(s).longValue() : Long.valueOf(s);
+        return s.matches("-?\\d+") ? Long.valueOf(s) : Double.valueOf(s).longValue();